compiler/parser.nim

```diff
@@ -91,7 +91,7 @@ proc closeParser(p: var TParser) =
 
 proc parMessage(p: TParser, msg: TMsgKind, arg = "") =
   ## Produce and emit the parser message `arg` to output.
-  lexMessage(p.lex, msg, arg)
+  lexMessageTok(p.lex, msg, p.tok, arg)
```

Comment:
Very good question. I think this is correct. I've looked into a parser and `errInvalidIndentation` is caused by wrong `p.tok.indent`, so the this error applies to the token itself, so we expect compiler point on the token, not `l.bufpos` which points on the 1st char after the token.
Actually parser always work on the token not on the characters in general, so it is okay that lexer use `l.bufpos` to determine problematic place, but for parser it is IMHO always location of token.